crates/holochain/src/conductor/conductor/tests.rs

```diff
@@ -1172,3 +1172,65 @@ async fn test_init_concurrency() {
     assert_eq!(num_calls_clone.fetch_add(0, Ordering::SeqCst), 100);
     assert_eq!(num_inits_clone.fetch_add(0, Ordering::SeqCst), 1);
 }
+
+/// Check that an app can be installed with deferred memproof provisioning and:
+/// - all status checks return correctly while still provisioned,
+/// - no zome calls can be made while awaiting provisioning,
+/// - app functions correctly after provisioning
+#[tokio::test(flavor = "multi_thread")]
+async fn test_deferred_provisioning() {
```

Comment:
I agree with these except for disabling clone cell. Adding a clone cell only adds it to the app definition, but doesn't actually create a clone at that moment, so I think this is fine. It could be valid for someone to set up the intended clones to run when the app is enabled, even though this is not intended.

Also, I checked, and enabling an app technically succeeds, but doesn't change the app status or do anything else. Think that's OK?

Reply:
Fair enough for creating clone cells, but if it's meant to behave that way I still think it would make sense to test that it does behave that way.

It's probably confusing for enable app to succeed when the app can't be enabled until memproofs are provided. I think it'd be clearer for that to return an error.

Reply:
I changed enable_app to produce an error in this case, but in the part 2 PR #3934